### PR TITLE
Implementation of a quality indicator after the name

### PR DESCRIPTION
**Pull request by Mrashes**
@wofsauge Implemented it as a number next to the name (its togglable if the user doesn't like it).  That said, I was unsure if this was the official quality as its coming from a lookup table.

Any feedback is appreciated, as I mentioned, first time with lua

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -242,8 +243,12 @@ function EID:printDescription(desc)
 				curName = curName.." ("..englishName..")"
 			end
 		end
+		local withQual = curName
```
Comment (wofsauge) [inline main.lua:246]: Why a new variable? just overwrite "curName" directly

File: main.lua [hunk 2]
```diff
@@ -231,6 +231,7 @@ function EID:printDescription(desc)
 	--Display Itemname
 	if EID.Config["ShowItemName"] then
 		local curName = desc.Name
+		local quality = EID.itemWeightsLookup[tonumber(desc.ID)]
```
Comment (wofsauge) [inline main.lua:234]: that will throw errors on modded items. Lookup table is fine, but should be avoided when modded items are potential targets. So please read the quality from the game directly: https://wofsauge.github.io/IsaacDocs/rep/ItemConfig_Item/#quality

File: eid_config.lua [hunk 3]
```diff
@@ -175,7 +175,10 @@ EID.UserConfig = {
 	["ErrorColor"] = "ColorEIDError",
 	-- Enables or disables custom Descriptions for entities
 	-- Default = true
-	["EnableEntityDescriptions"] = true
+	["EnableEntityDescriptions"] = true,
+	-- Enables or disables item quality after item name
+	-- Default = true
+	["ShowQuality"] = true
```
Comment (wofsauge) [inline eid_config.lua:181]: this new option also needs to be added to the default config at the bottom of the config file.

It also needs a new mod menu entry (just copy paste stuff around in the mod config menu file ;) )

**Review by wofsauge — CHANGES_REQUESTED** [review]
generally good implementation, but has some flaws

**Comment by LuanRoger** [comment]
@wofsauge Is it possible to create an interpreter that would receive the quality number and return a string according to the quality (int)?
Like:
0 = Common 
1 = Rare etc

I think it would be better.

**Comment by wofsauge** [comment]
> @wofsauge Is it possible to create an interpreter that would receive the quality number and return a string according to the quality (int)?
> Like:
> 0 = Common
> 1 = Rare etc
> 
> I think it would be better.

My prefered way would be to have some icon for each quality. Text would be cool too i think...

How about an option to switch between display modes as well?

**Comment by LuanRoger** [comment]
> > @wofsauge Is it possible to create an interpreter that would receive the quality number and return a string according to the quality (int)?
> > Like:
> > 0 = Common
> > 1 = Rare etc
> > 
> > I think it would be better.
> 
> My prefered way would be to have some icon for each quality. Text would be cool too i think...
> 
> How about an option to switch between display modes as well?
> 

I could do it by text because it has the option of putting color markup like {{Yellow}} or {{Rainbow}}, like Terraria :)
But of course there could be an option to switch between them.

**Comment by wofsauge** [comment]
> > > @wofsauge Is it possible to create an interpreter that would receive the quality number and return a string according to the quality (int)?
> > > Like:
> > > 0 = Common
> > > 1 = Rare etc
> > > I think it would be better.
> > 
> > 
> > My prefered way would be to have some icon for each quality. Text would be cool too i think...
> > How about an option to switch between display modes as well?
> 
> I could do it by text because it has the option of putting color markup like {{Yellow}} or {{Rainbow}}, like Terraria :)
> But of course there could be an option to switch between them.

What colors and names would you assign to the qualities?
maybe like the WoW item quality system:
0 = Common = gray
1 = Uncommon = green
2 = Rare = blue
3 = Epic = purple
4 = Legendary = orange

(the names sound very dramatic and dont really represent what they mean in isaac environment imo :D )

**Comment by LuanRoger** [comment]
It's hard to say, but I think it's good to start (even not representing Issac), this is something that can be improved with ideas and suggestions from the community.

**Comment by wofsauge** [comment]
image
Ive added 5 icons for quality now that you can use if you want.

Icon names are {{Quality0}} to {{Quality4}}

**Comment by Mrashes** [comment]
Oh sweet, that looks amazing.  I don't have bandwidth today but will try to get it by Saturday.

File: main.lua (hunk 1 above)
Comment (Mrashes) [reply main.lua:246]: Came from wanting immutability but based on your implementation I'll refine

**Comment by Mrashes** [comment]
Adjustments implemented.  Should be better

**Comment by Mrashes** [comment]
On play testing, it now shows quality for all instances (sac room, cards, pills)  Do we want it to be this way or should I limit it?

**Comment by Mrashes** [comment]
also more of a bug.  With key begger and samsons ball next to each other its using different icons for quality.  unsure why

**Pull request merged**